src/main/java/io/reactivex/internal/operators/flowable/FlowableFlattenIterable.java

```diff
@@ -0,0 +1,436 @@
+        this.prefetch = prefetch;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void subscribeActual(Subscriber<? super R> s) {
+        if (source instanceof Callable) {
+            T v;
+
+            try {
+                v = ((Callable<T>)source).call();
+            } catch (Throwable ex) {
+                Exceptions.throwIfFatal(ex);
+                EmptySubscription.error(ex, s);
+                return;
+            }
+
+            if (v == null) {
+                EmptySubscription.complete(s);
+                return;
+            }
+
+            Iterator<? extends R> it;
+
+            try {
+                Iterable<? extends R> iter = mapper.apply(v);
+
+                it = iter.iterator();
+            } catch (Throwable ex) {
+                Exceptions.throwIfFatal(ex);
+                EmptySubscription.error(ex, s);
+                return;
+            }
+
+            FlowableFromIterable.subscribe(s, it);
+
+            return;
+        }
+        source.subscribe(new FlattenIterableSubscriber<T, R>(s, mapper, prefetch));
+    }
+
+    static final class FlattenIterableSubscriber<T, R>
+    extends BasicIntQueueSubscription<R>
+    implements Subscriber<T> {
+
+        /** */
+        private static final long serialVersionUID = -3096000382929934955L;
+
+        final Subscriber<? super R> actual;
+
+        final Function<? super T, ? extends Iterable<? extends R>> mapper;
+
+        final int prefetch;
+
+        final int limit;
+
+        final AtomicLong requested;
+
+        Subscription s;
+
+        Queue<T> queue;
+
+        volatile boolean done;
+
+        volatile boolean cancelled;
+
+        final AtomicReference<Throwable> error;
+
+        Iterator<? extends R> current;
+
+        int consumed;
+
+        int fusionMode;
+
+        public FlattenIterableSubscriber(Subscriber<? super R> actual,
+                Function<? super T, ? extends Iterable<? extends R>> mapper, int prefetch) {
+            this.actual = actual;
+            this.mapper = mapper;
+            this.prefetch = prefetch;
+            this.limit = prefetch - (prefetch >> 2);
+            this.error = new AtomicReference<Throwable>();
+            this.requested = new AtomicLong();
+        }
+
+        @Override
+        public void onSubscribe(Subscription s) {
+            if (SubscriptionHelper.validate(this.s, s)) {
+                this.s = s;
+
+                if (s instanceof QueueSubscription) {
+                    @SuppressWarnings("unchecked")
+                    QueueSubscription<T> qs = (QueueSubscription<T>) s;
+
+                    int m = qs.requestFusion(ANY);
+
+                    if (m == SYNC) {
+                        fusionMode = m;
+                        this.queue = qs;
+                        done = true;
+
+                        actual.onSubscribe(this);
+
+                        return;
+                    } else
+                        if (m == ASYNC) {
```

Comment:
Eclipse CTRL+I sometimes does this to `else if`. I'll fix these and the previous onces but wait for you to finish the review